apps/lvgl_scanner/src/main.c

```diff
@@ -0,0 +1,302 @@
+        lv_obj_align(label, LV_ALIGN_BOTTOM_RIGHT, 0, 0);
+        lv_obj_set_size(entry, LV_PCT(100), 60);
+    } else {
+        label = lv_obj_get_child(entry, 1);
+        lv_label_set_text_fmt(label, "%d dB", desc->rssi);
+        label = lv_obj_get_child(entry, 2);
+    }
+
+    lv_label_set_text_fmt(label, "%s", desc->prim_phy == BLE_HCI_LE_PHY_1M ? "1M" : "Coded");
+    switch (desc->sec_phy) {
+    case BLE_HCI_LE_PHY_1M:
+        lv_label_ins_text(label, LV_LABEL_POS_LAST, "/1M");
+        break;
+    case BLE_HCI_LE_PHY_2M:
+        lv_label_ins_text(label, LV_LABEL_POS_LAST, "/2M");
+        break;
+    case BLE_HCI_LE_PHY_CODED:
+        lv_label_ins_text(label, LV_LABEL_POS_LAST, "/Coded");
+        break;
+    default:
+        break;
+    }
+
+    /* TODO: Add name label */
+}
+
+static void
+add_new_entry(struct ble_gap_ext_disc_desc *desc)
+{
+    uint8_t id;
+
+    lv_obj_t *entry = lv_obj_create(entries_list);
+
+    fill_entry_labels(entry, desc, 1);
+
+    id = lv_obj_get_child_cnt(entries_list) - 1;
+    entry_data[id].last_rssi = desc->rssi;
+    memcpy(entry_data[id].addr_val, desc->addr.val, 6);
+}
+
+static void
+update_entry(struct ble_gap_ext_disc_desc *desc, uint8_t id)
+{
+    uint8_t rssi_change;
+    lv_obj_t *entry = lv_obj_get_child(entries_list, id);
+
+    fill_entry_labels(entry, desc, 0);
+
+    /* Sort the list if and update RSSI if it changed significantly */
+    rssi_change = abs(entry_data[id].last_rssi - desc->rssi);
+    if (rssi_change > 5) {
+        entry_data[id].last_rssi = desc->rssi;
+        sort_entries();
+    }
+}
+
+static uint8_t
+is_event_connectable(struct ble_gap_ext_disc_desc *desc)
+{
+    return desc->props & BLE_HCI_ADV_CONN_MASK;
+}
+
+static void
+handle_disc_report(struct ble_gap_ext_disc_desc *desc)
+{
+    uint8_t id;
+
+    /* TODO: Decide what to do with >=MAX_ENTRIES entries (list reset vs overwriting last entry) */
+    if (desc->rssi < -70 || lv_obj_get_child_cnt(entries_list) >= MAX_ENTRIES) {
+        return;
+    }
+
+    /* Filter out non-connectable advertising events */
+    if (!is_event_connectable(desc)) {
+        return;
+    }
+
+    id = get_entry_id_by_addr(desc->addr.val);
+
+    if (id == ENTRY_ID_NONE) {
+        add_new_entry(desc);
+        sort_entries();
+    } else {
+        update_entry(desc, id);
+    }
+}
+
+static int
+scan_event(struct ble_gap_event *event, void *arg)
+{
+    switch (event->type) {
+    /* advertising report has been received during discovery procedure */
+    case BLE_GAP_EVENT_EXT_DISC:
+        handle_disc_report(&event->ext_disc);
+        return 0;
+
+    default:
+        MODLOG_DFLT(ERROR, "Discovery event not handled\n");
+        return 0;
+    }
+}
+
+static void
+start_scan(void)
+{
+    struct ble_gap_ext_disc_params coded;
+    struct ble_gap_ext_disc_params uncoded;
+    uint8_t filter_duplicates = 0;
+    uint8_t filter_policy = 0;
+    uint8_t limited = 0;
+    coded.passive = 1;
+    coded.itvl = 500;
```

Comment:
use `BLE_GAP_SCAN_ITVL_MS` and `BLE_GAP_SCAN_WIN_MS` macros and configure for 100% duty cycle